contracts/modules/cover/Cover.sol

```diff
@@ -47,18 +43,65 @@ contract Cover is ICover, MasterAwareV2 {
   uint public constant STAKE_SPEED_UNIT = 100000e18;
   uint public constant PRICE_CURVE_EXPONENT = 7;
   uint public constant MAX_PRICE_PERCENTAGE = 1e20;
+  IQuotationData internal immutable quotationData;
 
+  /* ========== CONSTRUCTOR ========== */
 
-  constructor() {
+  constructor(IQuotationData _quotationData, IProductsV1 _productsV1) {
+    quotationData = _quotationData;
+    productsV1 = _productsV1;
   }
 
-  function initialize(ICoverNFT _coverNFT) public {
-    require(address(coverNFT) == address(0), "Cover: already initialized");
+  function initialize(address _coverNFT) public {
```

Comment:
Storing it as an address allows you to use Cover.sol in contracts which use older solidity versions which do not allow non-primitive return values such as interfaces or structs without using experimental features. At the same time it could very well be stored in contract bytecode since coverNFT will be immutable.